src/node_crypto.cc

```diff
@@ -3603,6 +3603,12 @@ bool PublicKeyCipher::Cipher(const char* key_pem,
   if (EVP_PKEY_CTX_set_rsa_padding(ctx, padding) <= 0)
     goto exit;
 
+  /* Reset md for signatures, we are doing raw RSA ops anyway */
+  if (EVP_PKEY_CTX_get_operation(ctx) == EVP_PKEY_OP_SIGN) {
+    if (EVP_PKEY_CTX_set_signature_md(ctx, nullptr) <= 0)
```

Comment:
I admit I don't understand why it's necessary to clear the signature hash function.  Can you explain?  Preferably with a longer, in-code comment? :-)